Generation/src/components/PileUpEdmFileReaderTool.cpp

```diff
@@ -0,0 +1,81 @@
+
+#include "PileUpEdmFileReaderTool.h"
+#include "datamodel/MCParticleCollection.h"
+#include "datamodel/GenVertexCollection.h"
+
+#include "GaudiKernel/IIncidentSvc.h"
+#include "GaudiKernel/Incident.h"
+#include "GaudiKernel/IEventProcessor.h"
+
+#include "GaudiKernel/DeclareFactoryEntries.h"
+
+#include "podio/EventStore.h"
+#include "podio/ROOTReader.h"
+
+#include "HepMC/GenEvent.h"
+
+DECLARE_TOOL_FACTORY(PileUpEdmFileReader)
+
+PileUpEdmFileReader::PileUpEdmFileReader(const std::string& type, const std::string& name, const IInterface* parent)
+    : GaudiTool(type, name, parent) {
+  declareInterface<IHepMCProviderTool>(this);
+  declareProperty("filename", m_filename = "input.root", "EDM file which contains the particle collections");
+  declareProperty("particleCollectionName", m_particleCollectionName = "allGenParticles");
+  declareProperty("vertexCollectionName", m_vertexCollectionName = "allGenVertices");
+}
+
+PileUpEdmFileReader::~PileUpEdmFileReader() { ; }
+
+StatusCode PileUpEdmFileReader::initialize() {
+  m_store = podio::EventStore();
+  m_reader = podio::ROOTReader();
```

Comment:
Ah yes, after our discussion I meant to remove it. I implemented a very un-generic algorithm instead:

https://github.com/HEP-FCC/FCCSW/pull/173/commits/a791b2edfeefc4a56010f93c53a4daa7bb2a7f4a#diff-f363a567fa59d2cf5e319fc5297de46a

But this way has the drawback of being way more flexible (all events need to be merged before simulation). Yes, we should discuss.